File: dsa_new/setmatrix_zero.py
class Solution:
    def setmatrix(self,matrix):
        row=set()
        col=set()
        r=len(matrix)
        c=len(matrix[0])
        for m in range(r):
            for n in range(c):
                if matrix[m][n]==0:
                    row.add(m)
                    col.add(n)
        for m in range(r):
            for n in range(c):
                if m in row or n in  col:
                    matrix[m][n]=0
        return matrix
s=Solution()
matrix = [
    [1,1,1],
    [1,0,1],
    [1,1,1]
]
print(s.setmatrix(matrix))

File: dsa_new/setMatrixZero.js
function setMatrix(matrix)
{
    let row=new Set()
    let col=new Set()
    let r=matrix.length
    let c=matrix[0].length
    for(let i=0;i<r;i++)
    {
        for(let j=0;j<c;j++)
        {
            if(matrix[i][j]==0)
            {
                row.add(i)
                col.add(j)
            }
        }
    }
    for(let i=0;i<r;i++)
    {
        for(let j=0;j<c;j++)
        {
            if(row.has(i) || col.has(j))
            {
                matrix[i][j]=0
            }
        }
    }
    return matrix;
}
matrix = [
    [1,1,1],
    [1,0,1],
    [1,1,1]
]
console.log(setMatrix(matrix))